sanic/server.py

```diff
@@ -291,17 +294,27 @@ def on_headers_complete(self):
                 self.request
             )
             if self._is_stream_handler:
-                self.request.stream = asyncio.Queue()
+                self.request.stream = StreamBuffer(
+                    self.request_buffer_queue_size
+                )
                 self.execute_request_handler()
 
     def on_body(self, body):
         if self.is_request_stream and self._is_stream_handler:
             self._request_stream_task = self.loop.create_task(
-                self.request.stream.put(body)
+                self.body_append(body)
             )
             return
         self.request.body_push(body)
 
+    async def body_append(self, body):
```

Comment:
Do you think we need to figure out a way to test this (the pausing and resuming of the `transport`?

Reply:
yeah, but a real integration test (or maybe an example) might be more helpful in this situation. and we probably should assume `pause_reading` and `resume_reading ` working as it is, https://docs.python.org/3/library/asyncio-protocol.html#asyncio.ReadTransport.pause_reading.